detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MisusedAlsoSpec.kt

```diff
@@ -0,0 +1,92 @@
+package io.gitlab.arturbosch.detekt.rules.style
+
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.test.assertThat
+import io.gitlab.arturbosch.detekt.test.compileAndLint
+import org.junit.jupiter.api.Test
+
+class MisusedAlsoSpec {
+    val subject = MisusedAlso(Config.empty)
+
+    @Test
+    fun `does not report when no also is used`() {
+        val code = """
+            class Test {
+                private var a = 5
+                
+                init {
+                    a.let { 
+                        it.plus(5)
+                        it.minus(10)
+                    }
+                }
+            }
+        """.trimIndent()
+        assertThat(subject.compileAndLint(code)).isEmpty()
+    }
+
+    @Test
+    fun `reports an also in init of class`() {
+        val code = """
+            class Test {
+                private var a = 5
+                
+                init {
+                    a.also { 
+                        it.plus(5)
+                    }
+                }
+            }
+        """.trimIndent()
+        assertThat(subject.compileAndLint(code)).hasSize(1)
+    }
+
+    @Test
+    fun `reports an also on nullable type`() {
+        val code = """
+            class Test {
+                private var a: Int? = 5
+                
+                init {
+                    a?.also { 
+                        it.plus(5)
+                    }
+                }
+            }
+        """.trimIndent()
+        assertThat(subject.compileAndLint(code)).hasSize(1)
+    }
+
+    @Test
+    fun `reports an also with lambda passed as Argument in parenthesis`() {
+        val code = """
+            class Test {
+                private var a: Int? = 5
+                
+                init {
+                    a?.also({ 
+                        it.plus(5)
+                    })
+                }
+            }
+        """.trimIndent()
+        assertThat(subject.compileAndLint(code)).hasSize(1)
+    }
+
+    @Test
+    fun `does not report if it is not used in also`() {
+        val code = """
+            class Test {
+                private var a: Int? = 5
+                private var b: Int = 0
+                
+                init {
+                    a?.also({ 
+                        b.plus(5)
+                    })
```

Comment:
copy-paste, parentheses are not idiomatic
```suggestion
                    a?.also { 
                        b.plus(5)
                    }
```

Reply:
I actually wanted to have at least one test case, where this unidiomatic variant is used, because in this case I cannot access the via the `lambdaArguments` but rather by using the `valueArguments` of the expression. Just in case (even though highly unlikely) there is some kind of regression or change here, it will be caught by the unit tests.

Reply:
Yep, I agree, nice one. The reason this one stood out, is because the one above this one is specifically testing for it. "A test should fail for only one reason" - is the usual mantra. This test can fail because the parentheses thing is broken OR because `it` is not used.

Reply:
Resolved